source/mainnet/smart-contracts/tutorials/wCCD/wCCD-front-end-set-up.rst

```diff
@@ -4,6 +4,324 @@
+        :width: 40 %
+
+.. dropdown:: Workflow 2 - Setup browser wallet connected to local node (click here)
+
+    Before you start this workflow, make sure you have:
+
+    - a running Concordium testnet node that fully caught up.
+
+    - have port forwarding enabled (This step is only required when you run your node on a remote server instead of locally. See the note below):
+
+    .. note::
+
+        When your node is running on a remote server your cloud provider usually gives you an option to ssh into it.
+        Add the following port forwarding rule to your method to ssh into your instance by running the command in another terminal.
+        The port 10001 on your localhost is forwarded to the port 10001 on your instance.
+
+        .. code-block:: console
+
+            $ssh -NL localhost:10001:<IP-address-of-your-instance>:10001 <username>@<host>
+
+    If you don't have a running testnet node or port forwarding enabled, the :ref:`piggy bank tutorial part 3 <piggy-bank-preparing>`
+    will guide you through these setup steps.
+
+    These prerequisites ensure that you have a testnet node reachable locally on port 10001. The browser wallet requires the
+    `Concordium JSON-RPC server <https://github.com/Concordium/concordium-json-rpc>`_
+    that points to your node. Clone the repository with the command:
+
+    .. code-block:: console
+
+        $git clone https://github.com/Concordium/concordium-json-rpc.git
+
+    Build and run the server as described in the README file of the
+    `JSON-RPC repository <https://github.com/Concordium/concordium-json-rpc>`_.
+
+    The final command that you execute to start the ``JSON-RPC server`` is as follows:
+
+    .. code-block:: console
+
+        $yarn start --port 9095 --nodeAddress 127.0.0.1 --nodePort 10001 --nodeTimeout 5000
+
+    Your ``JSON-RPC server`` is running on port `9095` and connects to your local node on port `10001`.
+    You are ready now to build the pre-MVP browser wallet from the source code.
+    Clone the repository with the command:
+
+    .. code-block:: console
+
+        $git clone https://github.com/Concordium/concordium-browser-wallet.git
+
+    You can build the browser wallet extension and load it
+    (``dist`` folder from the path `root/packages/browser-wallet`) into the
+    `Chrome browser <https://developer.chrome.com/docs/extensions/mv3/getstarted/#unpacked>`_ as
+    described in the README file of the
+    `browser wallet repository <https://github.com/Concordium/concordium-browser-wallet/tree/main/packages/browser-wallet>`_.
+
+    .. note::
+
+        Depending on the exact commit hash that you used to build your pre-MVP browser wallet, the
+        screenshots and setup steps might differ. The browser wallet holds the private keys corresponding
+        to the accounts of the user and a link that points to a ``JSON-RPC server``.
+
+    The next steps are based on the pre-MVP browser wallet from a git commit before 22.8.2022.
```

Comment:
It would be very nice if the wallet supports one-day different types of connections (local nodes, API connection to server + hosted nodes by Concordium, API connection to server + hosted nodes by third-party providers).

So far the most important connection option is supported (API connection to server + hosted nodes by Concordium), when and if we will support the other options in the future, I am not exactly sure but I can ask.

Reply:
Reply from Emil Kristiansen: yes in one of the next releases (after the MVP is released we will add the option for the user to key in an endpoint of their preference)